mava/components/jax/updating/terminators.py

```diff
@@ -14,5 +14,92 @@
 # limitations under the License.
 
 """Terminator component for Mava systems."""
+import abc
+from typing import Any, Type
 
-# TODO (Dries)
+import launchpad as lp
+from chex import dataclass
+
+from mava.components.jax.component import Component
+from mava.core_jax import SystemParameterServer
+
+
+class Terminator(Component):
+    @abc.abstractmethod
+    def __init__(
+        self,
+        config: Any,
+    ):
+        """_summary_
+
+        Args:
+            config : _description_.
+        """
+        self.config = config
+
+    @abc.abstractmethod
+    def on_parameter_server_run_loop_termination(
+        self, parameter_sever: SystemParameterServer
+    ) -> None:
+        """_summary_"""
+        pass
+
+    @staticmethod
+    def name() -> str:
+        """_summary_
+
+        Returns:
+            _description_
+        """
+        return "temination_condition"
+
+
+@dataclass
+class ParameterTerminatorConfig:
+    parameter_termination_key: str = "executor_steps"
+    parameter_termination_value: int = 5000
+
+
+class ParameterTerminator(Terminator):
+    def __init__(
+        self,
+        config: Any = ParameterTerminatorConfig(),
+    ):
+        """_summary_
+
+        Args:
+            config : _description_.
+        """
+        self.config = config
+
+        valid_options = [
+            "trainer_steps",
+            "trainer_walltime",
+            "evaluator_steps",
+            "evaluator_episodes",
+            "executor_episodes",
+            "executor_steps",
+        ]
+        assert self.config.parameter_termination_key in valid_options, (
+            "Please give a valid termination condition. "
+            + f"Current valid conditions are {valid_options}"
+        )
```

Comment:
Please rather make this an Exception than an assertion. See [here](https://stackoverflow.com/questions/15046695/why-use-assert-instead-of-exception-throwing) for example. Also, it can be formatted entirely with `f`

Reply:
Sorted, I just need to keep multiple strings due to mypy PEP errors.